
From: Michael Hunold <hunold@linuxtv.org>

- [V4L] mxb, dpc7146, hexium_orion, hexium_gemini: follow latest changes in
  saa7146 driver

Signed-off-by: Michael Hunold <hunold@linuxtv.org>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/media/video/dpc7146.c       |   10 +++++++---
 25-akpm/drivers/media/video/hexium_gemini.c |    9 +++++++--
 25-akpm/drivers/media/video/hexium_orion.c  |    9 +++++++--
 25-akpm/drivers/media/video/mxb.c           |   11 ++++++++---
 4 files changed, 29 insertions(+), 10 deletions(-)

diff -puN drivers/media/video/dpc7146.c~V4L-follow-changes-in-saa7146 drivers/media/video/dpc7146.c
--- 25/drivers/media/video/dpc7146.c~V4L-follow-changes-in-saa7146	2004-09-20 11:22:41.420354112 -0700
+++ 25-akpm/drivers/media/video/dpc7146.c	2004-09-20 11:22:41.437351528 -0700
@@ -79,8 +79,8 @@ static struct saa7146_extension_ioctls i
 
 struct dpc
 {
-	struct video_device	video_dev;
-	struct video_device	vbi_dev;
+	struct video_device	*video_dev;
+	struct video_device	*vbi_dev;
 
 	struct i2c_adapter	i2c_adapter;	
 	struct i2c_client	*saa7111a;
@@ -106,7 +106,11 @@ static int dpc_probe(struct saa7146_dev*
 	   video port pins should be enabled here ?! */
 	saa7146_write(dev, MC1, (MASK_08 | MASK_24 | MASK_10 | MASK_26));
 
-	saa7146_i2c_adapter_prepare(dev, &dpc->i2c_adapter, I2C_CLASS_TV_ANALOG, SAA7146_I2C_BUS_BIT_RATE_480);
+	dpc->i2c_adapter = (struct i2c_adapter) {
+		.class = I2C_CLASS_TV_ANALOG,
+		.name = "dpc7146",
+	};
+	saa7146_i2c_adapter_prepare(dev, &dpc->i2c_adapter, SAA7146_I2C_BUS_BIT_RATE_480);
 	if(i2c_add_adapter(&dpc->i2c_adapter) < 0) {
 		DEB_S(("cannot register i2c-device. skipping.\n"));
 		kfree(dpc);
diff -puN drivers/media/video/hexium_gemini.c~V4L-follow-changes-in-saa7146 drivers/media/video/hexium_gemini.c
--- 25/drivers/media/video/hexium_gemini.c~V4L-follow-changes-in-saa7146	2004-09-20 11:22:41.431352440 -0700
+++ 25-akpm/drivers/media/video/hexium_gemini.c	2004-09-20 11:22:41.438351376 -0700
@@ -78,7 +78,8 @@ static struct v4l2_queryctrl hexium_cont
 struct hexium
 {
 	int type;
-	struct video_device	video_dev;
+
+	struct video_device	*video_dev;
 	struct i2c_adapter	i2c_adapter;
 		
 	int 		cur_input;	/* current input */
@@ -250,7 +251,11 @@ static int hexium_attach(struct saa7146_
 	/* enable i2c-port pins */
 	saa7146_write(dev, MC1, (MASK_08 | MASK_24 | MASK_10 | MASK_26));
 
-	saa7146_i2c_adapter_prepare(dev, &hexium->i2c_adapter, I2C_CLASS_TV_ANALOG, SAA7146_I2C_BUS_BIT_RATE_480);
+	hexium->i2c_adapter = (struct i2c_adapter) {
+		.class = I2C_CLASS_TV_ANALOG,
+		.name = "hexium gemini",
+	};
+	saa7146_i2c_adapter_prepare(dev, &hexium->i2c_adapter, SAA7146_I2C_BUS_BIT_RATE_480);
 	if (i2c_add_adapter(&hexium->i2c_adapter) < 0) {
 		DEB_S(("cannot register i2c-device. skipping.\n"));
 		kfree(hexium);
diff -puN drivers/media/video/hexium_orion.c~V4L-follow-changes-in-saa7146 drivers/media/video/hexium_orion.c
--- 25/drivers/media/video/hexium_orion.c~V4L-follow-changes-in-saa7146	2004-09-20 11:22:41.432352288 -0700
+++ 25-akpm/drivers/media/video/hexium_orion.c	2004-09-20 11:22:41.439351224 -0700
@@ -68,8 +68,9 @@ static struct saa7146_extension_ioctls i
 struct hexium
 {
 	int type;
-	struct video_device	video_dev;
+	struct video_device	*video_dev;
 	struct i2c_adapter	i2c_adapter;	
+
 	int cur_input;	/* current input */
 };
 
@@ -237,7 +238,11 @@ static int hexium_probe(struct saa7146_d
 	saa7146_write(dev, DD1_STREAM_B, 0x00000000);
 	saa7146_write(dev, MC2, (MASK_09 | MASK_25 | MASK_10 | MASK_26));
 
-	saa7146_i2c_adapter_prepare(dev, &hexium->i2c_adapter, I2C_CLASS_TV_ANALOG, SAA7146_I2C_BUS_BIT_RATE_480);
+	hexium->i2c_adapter = (struct i2c_adapter) {
+		.class = I2C_CLASS_TV_ANALOG,
+		.name = "hexium orion",
+	};
+	saa7146_i2c_adapter_prepare(dev, &hexium->i2c_adapter, SAA7146_I2C_BUS_BIT_RATE_480);
 	if (i2c_add_adapter(&hexium->i2c_adapter) < 0) {
 		DEB_S(("cannot register i2c-device. skipping.\n"));
 		kfree(hexium);
diff -puN drivers/media/video/mxb.c~V4L-follow-changes-in-saa7146 drivers/media/video/mxb.c
--- 25/drivers/media/video/mxb.c~V4L-follow-changes-in-saa7146	2004-09-20 11:22:41.434351984 -0700
+++ 25-akpm/drivers/media/video/mxb.c	2004-09-20 11:22:41.440351072 -0700
@@ -128,8 +128,8 @@ static struct saa7146_extension_ioctls i
 
 struct mxb
 {
-	struct video_device	video_dev;
-	struct video_device	vbi_dev;
+	struct video_device	*video_dev;
+	struct video_device	*vbi_dev;
 
 	struct i2c_adapter	i2c_adapter;	
 
@@ -183,7 +183,12 @@ static int mxb_probe(struct saa7146_dev*
 	}
 	memset(mxb, 0x0, sizeof(struct mxb));	
 
-	saa7146_i2c_adapter_prepare(dev, &mxb->i2c_adapter, I2C_CLASS_TV_ANALOG, SAA7146_I2C_BUS_BIT_RATE_480);
+	mxb->i2c_adapter = (struct i2c_adapter) {
+		.class = I2C_CLASS_TV_ANALOG,
+		.name = "mxb",
+	};
+
+	saa7146_i2c_adapter_prepare(dev, &mxb->i2c_adapter, SAA7146_I2C_BUS_BIT_RATE_480);
 	if(i2c_add_adapter(&mxb->i2c_adapter) < 0) {
 		DEB_S(("cannot register i2c-device. skipping.\n"));
 		kfree(mxb);
_
